### PR TITLE
Add links section and css

### PR DESCRIPTION
**Pull request by OniiCoder**
What is contained in this PR:

- config for links
- generated css for links
- link added to editor with controls

### CODE REVIEW

File: app/src/config.ts [hunk 1]
```diff
@@ -45,6 +45,21 @@ const light = signal({
     fontSize: '0.9em',
     padding: '0.5em 0.75em',
   },
+  link: {
```
Comment (blackmann) [inline app/src/config.ts:48]: Let’s add `fontWeight`.

File: app/src/sections/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            <option value="underline">Underline</option>
+            </Select>
+      </Fieldset>
+      
+      <Fieldset label=":link color">
+        <ColorPicker
+          onChange={(value) => {
+            setStyle('link.default.color', value)
+          }}
+          value={getStyle('link.default.color')}
```
Comment (blackmann) [inline app/src/sections/Link.tsx:57]:
```suggestion
          value={getStyle('link.default.color') || getStyle(‘body.color’)}
```

File: app/src/config.ts [hunk 3]
```diff
@@ -45,6 +45,21 @@ const light = signal({
...
+    textDecoration: 'none',
+    default: {
+      color: '#222',
+    },
+    hover: {
+      color: '#222',
+    },
+    active: {
+      color: '#222',
+    },
```
Comment (blackmann) [inline app/src/config.ts:58]: Leave these empty. Then use the body color as the default see [this comment](https://github.com/blackmann/stylebender/pull/32/files#r1252302825) in `Link.tsx` for what I mean.

```suggestion
    default: undefined,
    hoverColor: undefined // hover changes cursor to pointer, so it’s not really critical to have a different color on hover, what do you think?
```


Let’s do away with `:active` for now and set `textDecoration` as `underlined

File: app/src/sections/Link.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      </Fieldset>
+      
+      <Fieldset label=":visited color">
+        <ColorPicker
+          onChange={(value) => {
+            setStyle('link.visited.color', value)
+          }}
+          value={getStyle('link.visited.color')}
+        />
+      </Fieldset>
```
Comment (blackmann) [inline app/src/sections/Link.tsx:86]: Let’s ignore visited and active states for now

File: app/src/sections/Link.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    >
+      <Preview />
+    </Section>
+  )
+}
+
+function Preview() {
+  return (
+    <>
+      <ComponentHeader>:link</ComponentHeader>
```
Comment (blackmann) [inline app/src/sections/Link.tsx:28]:
```suggestion
      <ComponentHeader>link</ComponentHeader>
```

File: app/src/css.ts [hunk 6]
```diff
@@ -256,6 +256,29 @@ function input() {
   return [base, dark, invertCalendarIcon]
 }
 
+function link() {
+  const createStyle = (selector: string, colorConfigPath: string, mode = <Mode>'light') => {
```
Comment (blackmann) [inline app/src/css.ts:260]:
```suggestion
  const createStyle = (selector: string, colorConfigPath: string, mode: Mode = 'light') => {
```

File: app/src/css.ts [hunk 7]
```diff
@@ -256,6 +256,29 @@ function input() {
...
+    const style = new Style(`:is(.a, a)${selector}`, mode);
+    style.add('color', mode === 'light' ? l(colorConfigPath) : d(colorConfigPath));
+    style.add('cursor', 'pointer');
+    style.add('text-decoration', mode === 'light' ? l('link.textDecoration') : d('link.textDecoration'));
```
Comment (blackmann) [inline app/src/css.ts:264]: You’re adding `text-decoration` and `cursor` to every state but it should only appear on the main `<a />` element.

Get what I mean?

File: app/src/config.ts (hunk 1 above)
Comment (OniiCoder) [reply app/src/config.ts:48]: Done

File: app/src/config.ts (hunk 3 above)
Comment (OniiCoder) [reply app/src/config.ts:58]: I think you have a point, removed it.

File: app/src/css.ts (hunk 7 above)
Comment (OniiCoder) [reply app/src/css.ts:264]: Yeah I get you, checkout the changes now

File: app/src/css.ts [hunk 8]
```diff
@@ -256,6 +256,27 @@ function input() {
   return [base, dark, invertCalendarIcon]
 }
 
+function link() {
+  const base = new Style(':is(.a, a)')
```
Comment (blackmann) [inline app/src/css.ts:260]:
```suggestion
  const base = new Style('a')
```

Just 'a' is fine

File: app/src/css.ts [hunk 9]
```diff
@@ -256,6 +256,27 @@ function input() {
...
+  base.add('cursor', 'pointer')
+  base.add('text-decoration', 'underline')
+
+  const createStyle = (mode: Mode = 'light') => {
+    const style = new Style(`:is(.a, a)`, mode);
+    style.add('color', mode === 'light' ? l('link.defaultColor') : d('link.defaultColor'));
+    style.add('font-weight', mode === 'light' ? l('link.fontWeight') : d('link.fontWeight'));
+    return style;
+  };
```
Comment (blackmann) [inline app/src/css.ts:269]: I understand you're trying to DRY here... but this is creating 3 style blocks for `a` in the output

One with the cursor and text-decoration [static] values and another with color and font-weight. However, we'll want the output to look like:

```css
a {
  cursor: pointer;
  text-decoration: underline;
  color: red;
  font-weight: medium;
}
```

**Comment by blackmann** [comment]
Please address this build error too:

Screenshot 2023-07-05 at 9 01 56 AM

I think we're close to merge! Thanks man

**Comment by OniiCoder** [comment]
@blackmann kindly check again.

**Review by blackmann — APPROVED** [review]

**Pull request merged**